File: HighAndLowS07.java
package jp.co.f1.superintro.ch8;

import java.util.Scanner;				//キーボード入力をする為のimport文Scannerクラス準備

public class HighAndLowS07{

	public static void main(String[] args) {
		Scanner sc=new Scanner(System.in);				//Scannerクラスの変数sc宣言
		System.out.println("**************");			//タイトル表示
		System.out.println("* High & Low *");
		System.out.println("**************");
		System.out.println();
		while(true) {														//ループさせる為のwhile文
			int leftCard=(int)(Math.random()*9)+1;		//int型変数leftCardにランダムで1～９の数値を代入
			int rightCard=(int)(Math.random()*9)+1;	//int型変数rightCardにランダムで1～９の数値を代入
			System.out.println("   [問題表示]");				//問題表示と左のカードと右の伏せカード表示
			System.out.println("*****    *****");
			System.out.println("*   *    * * *");
			System.out.println("* "+leftCard+" *    * * *");
			System.out.println("*   *    * * *");
			System.out.println("*****    *****");
			System.out.print("High or Low ?(h/l) >");			//h or l入力促す文を表示
			String select=sc.nextLine();			//String型に対応したnextLine()メソッドを使用してキーボード入力画面追加
			if(select.equals("h")){				//HighまたはLowを選択しました。を表示
				System.out.println("→Highを選択しました。");				//キーボード入力した文字がhならHighを選択しました。を表示
			}else {
				System.out.println("→Lowを選択しました。");				//キーボード入力した文字がlならLowを選択しました。を表示
			}
			String result;														//勝敗判定の結果を入れる為のString型変数result宣言
			if(leftCard<rightCard) {				//勝敗判定if elseif文
				result="h";								//右のカードが左のカードより数字が大きかったらresultにhを代入
			}else if(leftCard>rightCard){
				result="l";								//右のカードが左のカードより数字が小さかったらresultにlを代入
			}else {
				result=select;							//右のカードと左のカードが同じ数字ならresultにselect(キーボード入力したhまたはl)を代入
			}
			System.out.println();								//結果表示と左と右のカードを表示
			System.out.println("   [結果表示]");
			System.out.println("*****    *****");
			System.out.println("*   *    *   *");
			System.out.println("* "+leftCard+" *    * "+rightCard+" *");
			System.out.println("*   *    *   *");
			System.out.println("*****    *****");
			if(result.equals(select)) {				//勝敗表示
				System.out.println("→You Win!");			//resultがキーボード入力したhまたはlと同じならYou Win!と表示してループ継続
				System.out.println();
			}else {
				System.out.println("→You Lose...");				//resultがキーボード入力したhまたはlと同じじゃなかったらYou Lose...と--ゲーム終了--を表示してbreakでループ終了
				System.out.println();
				break;
			}
			System.out.println("--ゲーム終了--");
		}
	}
}

File: highAndLowS07.php
<?php
//array関数を使ってトランプの画像名を配列で作成
$cards=array("Jk.png","01.png","02.png","03.png","04.png","05.png","06.png",
    "07.png","08.png","09.png","10.png","11.png","12.png","13.png");

//フォームのPOST送信データを各変数に代入
$lCard=$_POST['leftCard'];
$hlSelect=$_POST['select'];

//0～13の範囲でランダム数値を$rightCardに代入
$rightCard=mt_rand(0,13);
?>
<html>
	<head>
		<meta http-equiv="Content-Type"content="text/html;charset=UTF-8">
	</head>
	<body>
	<div align='center'>
	   <font size="6">High＆Lowゲーム</font><br><hr>
			<?php
		      echo'<img src="../cards/',$cards[$lCard],'">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
                &nbsp;&nbsp;&nbsp;&nbsp;'; //隠しフィールドで受け取った左カードと同じカードを表示
		      echo'<img src="../cards/',$cards[$rightCard],'">';  //ランダムで右のカードを表示
		      if($lCard<$rightCard){      //High or Lowを判定
		          $result="High";
		      }elseif($lCard>$rightCard){
		          $result="Low";
		      }else{
		          $result="$hlSelect";
		      }
	        ?>
	        <br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<br>
			<?php
			  echo '「',"$hlSelect",'」を選択しました。';  //「High」または「Low」を選択しました。を表示
			?><br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<br>
			<?php
			  if($hlSelect==$result){    //勝敗判定
			      echo 'You Win!';
			  }else{
			      echo 'You Lose...';
			  }
			?><br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<br>
			<?php
			     echo '<a href="http://localhost:8888/High_and_Low/highAndLowS03.php">
                    もう一度挑戦</a>'   //もう一度挑戦のリンクを表示
			?>
	</div>
	</body>
</html>
